FreeIT/src/lesson_10/Application.java

```diff
@@ -0,0 +1,237 @@
+package lesson_10;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.InputMismatchException;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Scanner;
+
+public class Application {
+
+	private Library libr;
+
+	public Application() {
+		super();
+		ArrayList<Book> book = new ArrayList<>();
+		Book book1 = new Book(123, "Atlant", "Roman");
+		Book book2 = new Book(345, "Puaro", "Detective");
+		Book book3 = new Book(567, "NLO", "Fantasy");
+		book.add(book1);
+		book.add(book2);
+		book.add(book3);
+		Library lib = new Library(book);
+		libr = lib;
+	}
+
+	public void start() {
+		menu();
+	}
+
+	private void menu() {
+		System.out.println(
+				"Выберите действие (введите число, соответствующее действию): \n1. Вывод всех книг.\n2. Добавление книги.\n3. Удаление книги.\n4. Редактирование книги.\n5. Выход.");
+		Scanner sc = new Scanner(System.in);
+		int choice = 0;
+		try {
+			choice = sc.nextInt();
+		} catch (InputMismatchException e) {
+			System.out.println("Вы должны ввести число! Попробуйте еще раз!");
+			menu();
+			return;
+		}
+		switch (choice) {
+		case 1:
+			printAllBooks();
+			menu();
```

Comment:
every iteration you adding additional stack to your "menu()" method. So you'll catch the StackOverflowError in a result. You don't need to use recursion here

Reply:
Please, give me a hint how to do it without recursion?

Reply:
You cal loop choises inside while( wannaExit) {...} and assign boolean wannaExit with, for example, typing 99 on choise iteration